test/com/wsscode/pathom3/connect/planner_test.cljc

```diff
@@ -336,6 +336,124 @@
                                           :key          :b,
                                           :query        '...}}}))))))
 
+
+(deftest compute-run-graph--statically-unreachable-nodes-not-created
+  (testing "Nodes with nested provides that are known to be statically unreachable given nested query requires are not created"
+    (is (= #:com.wsscode.pathom3.connect.planner
+           {:nodes                 {1 {::pco/op-name 'nested-1-resolver
+                                       ::pcp/expects   {:container {}}
+                                       ::pcp/input     {}
+                                       ::pcp/node-id   1}}
+            :index-ast             {:container {:type         :join
+                                                :dispatch-key :container
+                                                :key          :container
+                                                :query        [:item-1]
+                                                :children     [{:type         :prop
+                                                                :dispatch-key :item-1
+                                                                :key          :item-1}]}}
+            :index-resolver->nodes {'nested-1-resolver #{1}}
+            :index-attrs           {:container #{1}}
+            :root                  1}
+           (compute-run-graph
```

Comment:
would be nice to have another case here where there are multiple options, but user doesn't send the sub-query (so keep both).

Also, the case where the user send a sub-query, but the output doesn't have any nested detail (should also keep all options in this case).

Reply:
Thanks, those are both very useful cases